token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,216 @@
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption
+                )
+
+            for param in response.get('Parameters', []):
+                parameters[param['Name']] = param['Value']
+
+            next_token = response.get('NextToken')
+            if not next_token:
+                break
+
+        return parameters
+    except(ssm.exceptions.InternalServerError, ssm.exceptions.InvalidFilterKey, ssm.exceptions.InvalidFilterOption,
+           ssm.exceptions.InvalidFilterValue, ssm.exceptions.InvalidKeyId, ssm.exceptions.InvalidNextToken) \
+            as ssm_exception:
+        print(f'ssm get_parameters_by_path error with path: {path} with_decryption:{with_decryption} '
+              f'error: {ssm_exception}')
+        raise ssm_exception
+    except Exception as e:
+        print(f'Unexpected error in get_parameters_by_path with path: {path} with_decryption:{with_decryption} '
+              f'error: {e}')
+        raise e
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    if not lambda_arn:
+        return "Error: lambda_arn not provided"
+
+    try:
+        input_params = json.loads(input_params_json)
+    except json.JSONDecodeError:
+        return f'Error: Invalid JSON : {input_params_json}'
+
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=json.dumps(input_params).encode('utf-8')
+        )
+
+        payload = response['Payload'].read().decode('utf-8')
+
+        # Check for function error
+        if 'FunctionError' in response:
+            return f"Error: Lambda function error: {payload}"
+
+        return payload
+
+    except Exception as e:
+        return f"Error: Lambda invocation failed: {e}"
+
+def validate_input(client_id:str, minimum_alive_secs:int) -> str:
+    err_msg=''
+    if client_id is None or client_id.strip() == '':
+        err_msg='Error: client_id is required'
+
+    pattern = re.compile(r'^[a-zA-Z0-9]{3,32}$')
+    if client_id and not pattern.match(client_id):
+        err_msg='client_id must be between length 3-32 with pattern [a-zA-Z0-9]{3,32}'
+
+    if minimum_alive_secs is not None and (not isinstance(minimum_alive_secs, int)):
+        err_msg = f'{err_msg}\n Minimum alive interval must be an integer'
+
+    if isinstance(minimum_alive_secs, int) and (minimum_alive_secs > 3300 or minimum_alive_secs < 0):
+        err_msg = f'{err_msg}\n Minimum alive interval must be an integer between 1 and 3300'
+    return err_msg
```

Comment:
If there's multiple errors with the inputs, each prior error message gets clobbered by the next one. We either need to return from this function after the first error detected, or put all the errors into a list and return the list